joblib/test/data/create_numpy_pickle.py

```diff
@@ -30,19 +30,52 @@ def get_joblib_version(joblib_version=joblib.__version__):
     return '.'.join([m.group(1) for m in matches if m is not None])
 
 
-def write_test_pickle(to_pickle):
+def write_test_pickle(to_pickle, args):
+    kwargs = {}
+    compress = args.compress
+    method = args.method
     joblib_version = get_joblib_version()
     py_version = '{0[0]}{0[1]}'.format(sys.version_info)
     numpy_version = ''.join(np.__version__.split('.')[:2])
-    print('file:', np.__file__)
-    pickle_filename = 'joblib_{0}_compressed_pickle_py{1}_np{2}.gz'.format(
-        joblib_version, py_version, numpy_version)
-    joblib.dump(to_pickle, pickle_filename, compress=True)
-    pickle_filename = 'joblib_{0}_pickle_py{1}_np{2}.pkl'.format(
-        joblib_version, py_version, numpy_version)
-    joblib.dump(to_pickle, pickle_filename, compress=False)
+
+    # The game here is to generate the right filename according to the options.
+    body = '_compressed' if (compress and method == 'zlib') else ''
+    if compress:
+        if method == 'zlib':
+            kwargs['compress'] = True
+            extension = '.gz'
+        else:
+            kwargs['compress'] = (method, 3)
+            extension = '.pkl.{0}'.format(method)
+        if args.cache_size:
+            kwargs['cache_size'] = 0
+            body += '_cache_size'
+    else:
+        extension = '.pkl'
+
+    pickle_filename = 'joblib_{0}{1}_pickle_py{2}_np{3}{4}'.format(
+        joblib_version, body, py_version, numpy_version, extension)
+
+    try:
+        joblib.dump(to_pickle, pickle_filename, **kwargs)
+    except Exception as e:
+        print("Error: cannot generate file '{0}' with arguments '{1}'. "
+              "Error was: {2}".format(pickle_filename, kwargs, e))
+        sys.exit(1)
```

Comment:
It was to avoid stracktrace displayed when calling this python script via the whole data generation bash script for all available python version and compressors. I removed the `sys.exit(1)` and use an `else:` statement instead.